presto-main/src/main/java/com/facebook/presto/server/security/ServerSecurityModule.java

```diff
@@ -29,9 +30,30 @@
     @Override
     protected void setup(Binder binder)
     {
-        configBinder(binder).bindConfig(SecurityConfig.class);
+        configBinder(binder).bindConfig(LdapServerConfig.class);
+        LdapServerConfig ldapServerConfig = buildConfigObject(LdapServerConfig.class);
 
+        configBinder(binder).bindConfig(SecurityConfig.class);
         SecurityConfig config = buildConfigObject(SecurityConfig.class);
+
+        checkState(!(ldapServerConfig.getAuthenticationEnabled() && config.getAuthenticationEnabled()), "Enabling both Kerberos and LDAP authentication not supported");
```

Comment:
Rather than doing this, we should remove the `http.server.authentication.enabled` property from `SecurityConfig` (move it to `@DefunctConfig` -- see other usages for an example) and replace it with `http.server.authentication.type` that has values `none`, `ldap`, `kerberos`.

Also, `SecurityConfig` should only have the `http.server.authentication.type` property. The Kerberos properties should be moved out into a separate config class.